nixos/modules/services/web-apps/alps.nix

```diff
@@ -0,0 +1,99 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.alps;
+in {
+  options.services.alps = {
+    enable = mkEnableOption "alps";
+
+    port = mkOption {
+      type = types.port;
+      default = 1323;
+      description = ''
+        TCP port the process should listen to.
+      '';
+    };
+
+    bindIP = mkOption {
+      default = "[::]";
+      type = types.str;
+      description = ''
+        IP the service should listen to.
+      '';
+    };
+
+    theme = mkOption {
+      type = types.enum [ "alps" "sourcehut" ];
+      default = "sourcehut";
+      description = ''
+        The frontend's theme.
+      '';
+    };
+
+    imaps = {
+      port = mkOption {
+        type = types.port;
+        default = 993;
+        description = ''
+          The imaps port of the server
+        '';
+      };
+
+      host = mkOption {
+        type = types.str;
+        default = "[::1]";
+        example = "mail.example.org";
+        description = ''
+          The imaps server's address.
+        '';
+      };
+    };
+
+    smtps = {
+      port = mkOption {
+        type = types.port;
+        default = 445;
+        description = ''
+          The smtps port of the server
+        '';
+      };
+
+      host = mkOption {
+        type = types.str;
+        default = cfg.imaps.host;
+        defaultText = "services.alps.imaps.host";
+        example = "mail.example.org";
+        description = ''
+          The smtps server's address.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.alps = {
+      description = "alps is a simple and extensible webmail.";
+      documentation = [ "https://git.sr.ht/~migadu/alps" ];
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target" ];
+
+      path = with pkgs; [ coreutils ];
+
+      script = ''
+        exec ${pkgs.alps}/bin/alps \
+          -addr ${cfg.bindIP}:${toString cfg.port}\
+          -theme ${cfg.theme}\
+          imaps://${cfg.imaps.host}:${toString cfg.imaps.port}\
+          smpts://${cfg.smtps.host}:${toString cfg.smtps.port}
+      '';
+
+      serviceConfig = {
+        StateDirectory = "alps";
+        WorkingDirectory = pkgs.alps.public;
```

Comment:
The matter with that is that the software is searching for the themes and the plugins in its working directory. (Yes I know, this should™ be configurable)
Copying the respective directories to `/var/lib/` did not work out because after restarting the service, systemd's ephemeral user changed and the service did ofcourse not start due to missing permissions on the directories.
A solution to that would be appreciated